explorer/e2e/testFunctions.ts

```diff
@@ -301,27 +301,39 @@ export async function testFilterPresence(
  * @param filterOptionName - the name of the filter option
  * @returns a Playwright locator to the filter button
  */
-export const getNamedFilterButtonLocator = (
+export const getNamedFilterOptionLocator = (
   page: Page,
   filterOptionName: string
 ): Locator => {
-  return page
-    .getByRole("button")
-    .filter({ has: page.getByRole("checkbox"), hasText: filterOptionName });
+  return page.getByRole("button").filter({
+    has: page.getByRole("checkbox"),
+    hasText: RegExp(`^${escapeRegExp(filterOptionName)}\\s*\\d+\\s*`),
```

Comment:
Would you be able to add comments describing the why behind the regex pattern here?